pkg/deploy/reaper/reaper.go

```diff
@@ -57,10 +57,13 @@ func (reaper *DeploymentConfigReaper) Stop(namespace, name string, gracePeriod *
 		// so we shouldn't error out here
 		glog.V(2).Info(err)
 	}
-	if err = reaper.kc.ReplicationControllers(namespace).Delete(util.LatestDeploymentNameForConfig(dc)); err != nil {
-		// The deploymentConfig may not have a replication controller to delete
-		// so we shouldn't error out here
-		glog.V(2).Info(err)
+	// Remove all the deployments of the configuration
+	for version := 1; version <= dc.LatestVersion; version++ {
```

Comment:
Instead, perform a label query for RCs where `DeploymentConfigAnnotation=dc.Name`.